code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -51,91 +60,153 @@
 /obj/item/clothing/suit/armor/reactive/teleport
 	name = "reactive teleport armor"
 	desc = "Someone separated our Research Director from his own head!"
+	emp_message = "<span class='warning'>The reactive armor's teleportation calculations begin spewing errors!</span>"
 	var/tele_range = 6
 	var/rad_amount= 15
-	reactivearmor_cooldown_duration = 100
+	reactivearmor_cooldown_duration = 10 SECONDS
 
-/obj/item/clothing/suit/armor/reactive/teleport/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
-	if(!active)
-		return FALSE
-	if(prob(hit_reaction_chance))
-		if(world.time < reactivearmor_cooldown)
-			owner.visible_message("<span class='danger'>The reactive teleport system is still recharging! It fails to teleport [owner]!</span>")
-			return FALSE
-		owner.visible_message("<span class='danger'>The reactive teleport system flings [owner] clear of [attack_text], shutting itself off in the process!</span>")
-		playsound(get_turf(owner),'sound/magic/blink.ogg', 100, TRUE)
-		do_teleport(owner, get_turf(owner), tele_range, no_effects = TRUE, channel = TELEPORT_CHANNEL_BLUESPACE)
-		owner.rad_act(rad_amount)
-		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
-		return TRUE
-	return FALSE
+/obj/item/clothing/suit/armor/reactive/teleport/reactive_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>The reactive teleport system flings [owner] clear of [attack_text], shutting itself off in the process!</span>")
+	playsound(get_turf(owner),'sound/magic/blink.ogg', 100, TRUE)
+	do_teleport(owner, get_turf(owner), tele_range, no_effects = TRUE, channel = TELEPORT_CHANNEL_BLUESPACE)
+	owner.rad_act(rad_amount)
+	reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
+	return TRUE
+
+/obj/item/clothing/suit/armor/reactive/teleport/emp_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>The reactive teleport system ALMOST flings [owner] clear of [attack_text], shutting itself off and leaving something behind in the process!</span>")
+	var/drop_organ = prob(50)
+	if(drop_organ)
+		owner.spew_organ(0)
+	else
+		var/obj/item/bodypart/body_part = owner.get_bodypart(pick(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM, BODY_ZONE_R_LEG, BODY_ZONE_L_LEG))
+		if(body_part)
+			body_part.drop_limb(FALSE, TRUE)
+	playsound(get_turf(owner),'sound/machines/buzz-sigh.ogg', 50, TRUE)
+	playsound(get_turf(owner),'sound/magic/blink.ogg', 100, TRUE)
+	do_teleport(owner, get_turf(owner), tele_range, no_effects = TRUE, channel = TELEPORT_CHANNEL_BLUESPACE)
+	owner.rad_act(rad_amount)
```

Comment:
maybe increase the rads emitted by the armor here? to, say, 5x rad_amount, assuming that that doesn't just ensure your death if you don't have anti-rads meds or something